gordon_janitor_gcp/gce_instance_authority.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A GCEInstanceAuthority object retrieves a list of all instances in all
+projects that the ``keyfile`` gives it access to. It is meant to work
+in an asynchronous context and to be executed through the ``start()`` method.
+
+It is also meant to be easy to extend and customize so all methods except
+``start`` can be overridden as long as they follow the interfaces over the
+base methods. For example, if a different project filtering scheme is to be
+used, it's possible to inherit from GCEInstanceAuthority and override
+the ``get_all_projects`` method and ensure it returns a set of project IDs.
+
+To use:
+
+.. code-block:: pycon
+
+    >>> import asyncio
+    >>> rrset_channel = asyncio.Queue()
+    >>> authority = GCEInstanceAuthority(config, rrset_channel)
+    >>> await authority.start()
+    >>> msg = await.channel.get()
```

Comment:
is this the correct syntax? i'm not entirely familiar with asyncio api, but it looks a bit odd